### PR TITLE
soc: align CONFIG_SOC values to match soc.yml names

### PR DESCRIPTION
**Pull request by tejlmand**
This commit aligns CONFIG_SOC values for Nordic nRF SoCs to the names defined in soc.yml.

### CODE REVIEW

File: soc/nordic_nrf/nrf51/Kconfig.soc [hunk 1]
```diff
@@ -21,6 +21,6 @@ config SOC_NRF51822_QFAC
 	select SOC_SERIES_NRF51X
 
 config SOC
-	default "nRF51822_QFAA" if SOC_NRF51822_QFAA
-	default "nRF51822_QFAB" if SOC_NRF51822_QFAB
-	default "nRF51822_QFAC" if SOC_NRF51822_QFAC
+	default "nrf51822" if SOC_NRF51822_QFAA
+	default "nrf51822" if SOC_NRF51822_QFAB
+	default "nrf51822" if SOC_NRF51822_QFAC
```
Comment (nordicjm) [inline soc/nordic_nrf/nrf51/Kconfig.soc:26]:
```suggestion
	default "nrf51822" if SOC_NRF51822_QFAA || SOC_NRF51822_QFAB || SOC_NRF51822_QFAC
```

File: soc/nordic_nrf/nrf52/Kconfig.soc [hunk 2]
```diff
@@ -81,14 +81,14 @@ config SOC_NRF52840_QIAA
...
-	default "nRF52833_QIAA" if SOC_NRF52833_QIAA
-	default "nRF52840_QFAA" if SOC_NRF52840_QFAA
-	default "nRF52840_QIAA" if SOC_NRF52840_QIAA
+	default "nrf52805" if SOC_NRF52805_CAAA
+	default "nrf52810" if SOC_NRF52810_QFAA
+	default "nrf52811" if SOC_NRF52811_QFAA
+	default "nrf52820" if SOC_NRF52820_QDAA
+	default "nrf52832" if SOC_NRF52832_CIAA
+	default "nrf52832" if SOC_NRF52832_QFAA
+	default "nrf52832" if SOC_NRF52832_QFAB
```
Comment (nordicjm) [inline soc/nordic_nrf/nrf52/Kconfig.soc:90]:
```suggestion
	default "nrf52832" if SOC_NRF52832_CIAA || SOC_NRF52832_QFAA || SOC_NRF52832_QFAB
```

File: soc/nordic_nrf/nrf52/Kconfig.soc [hunk 3]
```diff
@@ -81,14 +81,14 @@ config SOC_NRF52840_QIAA
...
+	default "nrf52833" if SOC_NRF52833_QDAA
+	default "nrf52833" if SOC_NRF52833_QIAA
```
Comment (nordicjm) [inline soc/nordic_nrf/nrf52/Kconfig.soc:92]:
```suggestion
	default "nrf52833" if SOC_NRF52833_QDAA || SOC_NRF52833_QIAA
```

File: soc/nordic_nrf/nrf52/Kconfig.soc [hunk 4]
```diff
@@ -81,14 +81,14 @@ config SOC_NRF52840_QIAA
...
+	default "nrf52840" if SOC_NRF52840_QFAA
+	default "nrf52840" if SOC_NRF52840_QIAA
```
Comment (nordicjm) [inline soc/nordic_nrf/nrf52/Kconfig.soc:94]:
```suggestion
	default "nrf52840" if SOC_NRF52840_QFAA || SOC_NRF52840_QIAA
```

File: soc/nordic_nrf/nrf53/Kconfig.soc [hunk 5]
```diff
@@ -26,5 +26,5 @@ config SOC_NRF5340_CPUNET_QKAA
 	select SOC_NRF5340_CPUNET
 
 config SOC
-	default "nRF5340_CPUAPP_QKAA" if SOC_NRF5340_CPUAPP_QKAA
-	default "nRF5340_CPUNET_QKAA" if SOC_NRF5340_CPUNET_QKAA
+	default "nrf5340" if SOC_NRF5340_CPUAPP_QKAA
+	default "nrf5340" if SOC_NRF5340_CPUNET_QKAA
```
Comment (nordicjm) [inline soc/nordic_nrf/nrf53/Kconfig.soc:30]:
```suggestion
	default "nrf5340" if SOC_NRF5340_CPUAPP_QKAA || SOC_NRF5340_CPUNET_QKAA
```

File: soc/nordic_nrf/nrf52/Kconfig.soc (hunk 3 above)
Comment (tejlmand) [reply soc/nordic_nrf/nrf52/Kconfig.soc]: Done

File: soc/nordic_nrf/nrf51/Kconfig.soc (hunk 1 above)
Comment (tejlmand) [reply soc/nordic_nrf/nrf51/Kconfig.soc]: Done

File: soc/nordic_nrf/nrf52/Kconfig.soc (hunk 2 above)
Comment (tejlmand) [reply soc/nordic_nrf/nrf52/Kconfig.soc]: Done

File: soc/nordic_nrf/nrf52/Kconfig.soc (hunk 4 above)
Comment (tejlmand) [reply soc/nordic_nrf/nrf52/Kconfig.soc]: Done

File: soc/nordic_nrf/nrf53/Kconfig.soc (hunk 5 above)
Comment (tejlmand) [reply soc/nordic_nrf/nrf53/Kconfig.soc]: Done

**Review by nordicjm — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**